simulation/traffic_simulator/src/api/api.cpp

```diff
@@ -243,22 +243,40 @@ bool API::updateTrafficLightsInSim()
   return res.result().success();
 }
 
+bool API::updateNonEgoEntitiesStatusInSim()
+{
+  simulation_api_schema::UpdateEntityStatusRequest req;
+  req.set_npc_logic_started(entity_manager_ptr_->isNpcLogicStarted());
+
+  for (const auto & entity_name : entity_manager_ptr_->getEntityNames()) {
+    if (entity_manager_ptr_->isEgo(entity_name)) continue;
+    auto status = entity_manager_ptr_->getEntityStatus(entity_name);
+    simulation_interface::toProto(static_cast<EntityStatus>(status), *req.add_status());
+  }
+  simulation_api_schema::UpdateEntityStatusResponse res;
+  zeromq_client_.call(req, res);
+  return res.result().success();
+}
+
 std::optional<CanonicalizedEntityStatus> API::updateEntityStatusInSim(
   const std::string & entity_name, const CanonicalizedEntityStatus & status)
 {
   simulation_api_schema::UpdateEntityStatusRequest req;
-  simulation_api_schema::EntityStatus proto;
+  req.set_npc_logic_started(entity_manager_ptr_->isNpcLogicStarted());
   auto status_non_canonicalized = static_cast<EntityStatus>(status);
   status_non_canonicalized.name = entity_name;
-  simulation_interface::toProto(status_non_canonicalized, *req.mutable_status());
-  req.set_npc_logic_started(entity_manager_ptr_->isNpcLogicStarted());
+  simulation_interface::toProto(status_non_canonicalized, *req.add_status());
+
   simulation_api_schema::UpdateEntityStatusResponse res;
   zeromq_client_.call(req, res);
 
-  if (res.result().success()) {
-    simulation_interface::toMsg(res.status().pose(), status_non_canonicalized.pose);
+  if (res.result().success() && req.status_size() == 1) {
+    assert(
+      res.status(0).name() == entity_name &&
```

Comment:
Why do you hard-coded `0`?
Please remove hard-coded value if possible.
If you cannot remove this value, please leave doxygen comment and describe why the value should be `0`

Reply:
This method allows to call UpdateEnityStatusRequest for exactly one Entity (especially Ego) - for this reason I added a check if there is also only one status in the returned Response (check req.status_size() == 1) and perform operations only for this status (res.status(0).name())

Reply:
I improved the readability - added constexpr and provide a comment.